### PR TITLE
Created dummy communicator

### PR DESCRIPTION
**Pull request by oierlauzi**
Dummy communicator to used when a "real" communicator such as MPI is not available. It mocks a single rank communicator, such that most operations are no-ops or copies.

### CODE REVIEW

File: src/communication/dummy/dummy_communicator_backend.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+namespace communication
+{
+
+static const std::string name = "dummy";
+
+const std::string& dummy_communicator_backend::get_name() const noexcept
+{
+    return name;
+}
```
Comment (MartinSalinas98) [inline src/communication/dummy/dummy_communicator_backend.cpp:46]: Why define name and then return it as variable.

This was going to change in #98, so, i'd recommend following the same path here, as that PR will be merged really soon.
(alternatively, we could separate that into a different PR and follow it's own merge process in every repo)

File: src/communication/dummy/dummy_communicator_backend.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+version dummy_communicator_backend::get_version() const noexcept
+{
+    return get_core_version();
```
Comment (MartinSalinas98) [inline src/communication/dummy/dummy_communicator_backend.cpp:50]: Where is this defined?

**Review by MartinSalinas98 — CHANGES_REQUESTED** [review]

File: src/communication/dummy/dummy_communicator_backend.cpp (hunk 1 above)
Comment (oierlauzi) [reply src/communication/dummy/dummy_communicator_backend.cpp:46]: It is like that so that it compiles in the action (the changes from #98 are not merged). Depending on which one is merged before, it will be changed in this PR or in the other

File: src/communication/dummy/dummy_communicator_backend.cpp (hunk 2 above)
Comment (oierlauzi) [reply src/communication/dummy/dummy_communicator_backend.cpp:50]: core/core_version.hpp/cpp

**Review by MartinSalinas98 — APPROVED** [review]
